app/javascript/views/walletServices/braveConnection/upholdConnection/UpholdDisconnectPrompt.tsx

```diff
@@ -0,0 +1,59 @@
+import axios from "axios";
+import * as React from "react";
+import { FormattedMessage, injectIntl } from "react-intl";
+import routes from "../../../routes";
+
+import { LoaderIcon } from "brave-ui/components/icons";
+import { FlexWrapper } from "../../../style";
+
+const DisconnectPrompt = (props) => {
+  const [isLoading, setLoading] = React.useState(false);
+
+  const confirmDisconnect = async () => {
+    setLoading(true);
+    axios
+      .patch(routes.publishers.uphold.disconnect)
+      .then(() => {
+        setLoading(false);
+        props.setShowModal(false);
+        props.loadData();
+      })
+      .catch((response) => {
+        props.setError(response.data);
+      });
+  };
+
```

Comment:
Oh sweet you are using Hooks, consider Apollo & GraphQL for things similar to this